deepmd/dpmodel/descriptor/repformers.py

```
@@ -370,31 +400,31 @@ def call(
             g2 = g2 / self.rcut
             h2 = h2 / self.rcut
         # nf x nloc x nnei x ng2
         g2 = self.act(self.g2_embd(g2))
         # set all padding positions to index of 0
         # if a neighbor is real or not is indicated by nlist_mask
-        nlist[nlist == -1] = 0
+        nlist = xp.where(nlist == -1, xp.zeros_like(nlist), nlist)
         # nf x nall x ng1
-        mapping = np.tile(mapping.reshape(nf, -1, 1), (1, 1, self.g1_dim))
+        mapping = xp.tile(xp.reshape(mapping, (nf, -1, 1)), (1, 1, self.g1_dim))
         for idx, ll in enumerate(self.layers):
```

Ruff

Loop control variable `idx` not used within loop body

Rename unused `idx` to `_idx`

(B007)